AsyncFixer.Test/UnnecessaryAsyncTests.cs

```diff
@@ -201,9 +201,27 @@ static Task foo()
             VerifyCSharpFix(test, fixtest);
         }
 
-        // Task<object> ve Task<int> !!!
         [Fact]
-        public void NoWarn_UnnecessaryAsyncTest6()
+        public void GenericTask()
+        {
+            var test = @"
+using System;
+using System.Threading.Tasks;
+
+class Program
+{
+    async Task<int> foo(int i)
+    {
+        return await Task.Run(()=>3);
+    }
+}";
+
+            var expected = new DiagnosticResult { Id = DiagnosticIds.UnnecessaryAsync };
+            VerifyCSharpDiagnostic(test, expected);

```

Comment:
Wouldn't it be necessary here to verify the fix too?

Reply:
good catch!